src/protocols/rdp/channels/rdpdr/rdpdr-fs-messages.c

```diff
@@ -406,6 +430,9 @@ void guac_rdpdr_fs_process_query_directory(guac_rdp_common_svc* svc,
     if (file == NULL)
         return;
 
+    if (Stream_GetRemainingLength(input_stream) < 9)
```

Comment:
I recommend adding a comment noting the reason for doing this (that you're verifying the length of the PDU for future reads, and that the main header is 9 bytes). The check is otherwise initially mysterious and magic-numbery until the mental gears turn further. A comment noting that this number must line up with future reads would also serve to call attention to this section if any changes are made to those reads.

Reply:
Yeah, that makes sense.

Reply:
Added comments in that hopefully clarify this - let me know if they need any more detail.